### PR TITLE
Github CI

### PR DESCRIPTION
**Pull request by fosskers**
This PR introduces CI for the Angband project, ran via Github Actions. Each job takes only about 1 minute to run.

### TODO

- [x] Linux
  - [x] X11
  - [x] SDL
  - [x] NCurses
- [x] Mac (Cocoa)
- ~Windows~

### CODE REVIEW

**Comment by fosskers** [comment]
Since this is the first time Github Actions are being used in this repo, the jobs aren't being run yet. However you can see them succeed on [my forked PR](https://github.com/fosskers/angband/pull/1).

**Comment by fosskers** [comment]
For now I'm going to leave out the Windows build. The Github Action runners for Windows seem to want Powershell, not MinGW, etc., and I'm not familiar enough with Windows to experiment further.

**Comment by NickMcConnell** [comment]
OK, this is really cool.  I had planned to do this, but haven't got to it yet.  What I would really like to do, in fact, is do Mac and Windows builds of every push to master, and make them available to download.

So I'll try to get this pulled in soon, and we'll see where we go from there.

**Comment by fosskers** [comment]
It's totally possible to add that, so that's a great idea. I'm sure we can get the Windows build working too, as an extension of this initial work.

**Comment by fosskers** [comment]
Just now I noticed that there are test suites. Shall we run those too? What's the command for invoking them?

**Comment by NickMcConnell** [comment]
So I usually run the unit tests on the linux CLI with make tests, and the game start-up tests with ./run-tests

**Comment by fosskers** [comment]
Excellent, I'll add lines to that effect.

**Comment by fosskers** [comment]
I'm having some issues running the tests on my my own machine. Do you know what I'm doing wrong?
```
colin@yumi ~/c/c/angband (colin/github-ci)> pwd
/home/colin/code/c/angband
colin@yumi ~/c/c/angband (colin/github-ci)> make tests
Makefile:2: mk/buildsys.mk: No such file or directory
make: *** No rule to make target 'mk/buildsys.mk'.  Stop.
```
```
colin@yumi ~/c/c/angband (colin/github-ci) [2]> ./run-tests
Running: tests/birth/Dw-Pa... tests/run-test: line 14: src/angband: No such file or directory
Failed
Running: tests/birth/Hu-Wa... tests/run-test: line 14: src/angband: No such file or directory
Failed
Running: tests/birth/new-game-0... tests/run-test: line 14: src/angband: No such file or directory
Failed
Running: tests/birth/new-game-1... tests/run-test: line 14: src/angband: No such file or directory
Failed
Running: tests/trivial/matcher... tests/run-test: line 14: src/angband: No such file or directory
Passed
Total: 1/5
```

**Comment by NickMcConnell** [comment]
I usually run make from my top level angband directory; maybe the build directory is messing with it?

**Comment by fosskers** [comment]
Even after deleting `build/` the same thing happens. I'm trying to run both commands from the top level. Note that the error message from `run-tests` is telling the truth: there is no `src/angband` directory committed to the repo.

**Comment by NickMcConnell** [comment]
Maybe it's a configuration difference.  I configure to install, and I think the test suite is running the installed version.

**Comment by fosskers** [comment]
Ah that must be it - I've been following the [`cmake` instructions](https://angband.readthedocs.io/en/latest/hacking/compiling.html#compilation-with-cmake), and those are the ones I use in this PR. Let me try again with the approach that uses `configure`.

**Comment by fosskers** [comment]
Progress! `make tests` now works and passes. `./run-tests` gives me new errors now:
```
colin@yumi ~/c/c/angband (colin/github-ci)> ./run-tests
Running: tests/birth/Dw-Pa... src/angband: Unable to prepare any 'display module'!
Failed
Running: tests/birth/Hu-Wa... src/angband: Unable to prepare any 'display module'!
Failed
Running: tests/birth/new-game-0... src/angband: Unable to prepare any 'display module'!
Failed
Running: tests/birth/new-game-1... src/angband: Unable to prepare any 'display module'!
Failed
Running: tests/trivial/matcher... src/angband: Unable to prepare any 'display module'!
Passed
Total: 1/5
```

**Comment by NickMcConnell** [comment]
That is odd.  Can you run angband normally with x11?

**Comment by fosskers** [comment]
Yes, recently I play with `-mgcu` but when I started it was the default, which is X11 I believe. Either open fine for me.

**Comment by fosskers** [comment]
Ah but I suppose the `./run-tests` can't be run in CI anyway, given that they're not really run in a terminal. I'll adjust the Linux build here to run the unit tests, and then this PR should be good to go.

**Comment by fosskers** [comment]
That should do it!

**Comment by fosskers** [comment]
Wonderful!
